arangod/Replication2/StateMachines/Document/DocumentStateMethods.cpp

```diff
@@ -98,6 +97,32 @@ class DocumentStateMethodsDBServer final : public DocumentStateMethods {
     return leader;
   }
 
+  [[nodiscard]] auto getReplicationIterator(std::string_view shardId) const
+      -> ResultT<std::unique_ptr<ReplicationIterator>> {
+    auto logicalCollection = _vocbase.lookupCollection(shardId);
+    if (logicalCollection == nullptr) {
+      return ResultT<std::unique_ptr<ReplicationIterator>>::error(
+          TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND,
+          fmt::format("Collection {} not found", shardId));
+    }
+
+    auto ctx = transaction::StandaloneContext::Create(_vocbase);
+    auto singleCollectionTrx = SingleCollectionTransaction(
+        ctx, *logicalCollection, AccessMode::Type::READ);
+
+    // We call begin here so rocksMethods are initialized
+    if (auto res = singleCollectionTrx.begin(); res.fail()) {
+      return ResultT<std::unique_ptr<ReplicationIterator>>::error(
+          res.errorNumber(), res.errorMessage());
+    }
+
+    auto* physicalCollection = logicalCollection->getPhysical();
+    TRI_ASSERT(physicalCollection != nullptr);
+
+    return physicalCollection->getReplicationIterator(
+        ReplicationIterator::Ordering::Revision, singleCollectionTrx);
```

Comment:
This won't work like this. `getReplicationIterator` takes the `Methods`  via reference, but the `singleCollectionTrx` is going out of scope an will be destroyed.

Reply:
Thanks!